src/main/java/frc/robot/subsystems/drive/SwerveDrive.java

```diff
@@ -171,62 +176,60 @@ public SwerveDrive() {
     thetaPid.setTolerance(1);
 
     this.configurePathPlannerAutoBuilder();
+
+    SmartDashboard.putData(this);
   }
 
   /** {@inheritDoc} */
   @Override
-  public void periodic() {
-    updateOdometry();
-
-    var pose = getPose();
-    SmartDashboard.putNumber("Pose-X", pose.getX());
-    SmartDashboard.putNumber("Pose-Y", pose.getY());
-    SmartDashboard.putNumber("Pose-Theta", pose.getRotation().getDegrees());
+  public void initSendable(SendableBuilder builder) {
+    super.initSendable(builder);
+    builder.setSmartDashboardType(getName());
+    builder.setSafeState(this::stop);
+    builder.addDoubleProperty("Pose-X", () -> getPose().getX(), (xPoseMeters) -> {
```

Comment:
I believe you can write the getter something like the following.

```suggestion
    builder.addDoubleProperty("Pose-X", getPose()::getX, (xPoseMeters) -> {
```

Reply:
good call will do that

Reply:
Actually I tested this and I think it does not behave as I would think either. Using `getPose()::getX` gets the pose once and passes that value for X at the time AKA it does not get the new pose each time.